### PR TITLE
Tests hangs sometimes in IE11 when a js-error appeared on page load

### PR DESCRIPTION
**Pull request by AlexanderMoskovkin**

### CODE REVIEW

**Comment by AlexanderMoskovkin** [comment]
@testcafe-build-bot \retest

**Comment by AlexanderMoskovkin** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by inikulin** [comment]
@AlexanderMoskovkin FYI keeping just unstable tests doesn't work. BTW, you can filter log entries to the single browser, specify `DEBUG="testcafe:test-run:IE*:*"`

**Comment by AlexanderMoskovkin** [comment]
I have an idea about unstable tests but would like reproduce the problem before commiting the fix

**Comment by AlexanderMoskovkin** [comment]
@testcafe-build-bot \retest

**Comment by AlexanderMoskovkin** [comment]
@testcafe-build-bot \retest

**Comment by AlexanderMoskovkin** [comment]
@inikulin It seems the problem was in the following:
Particularly in the end of the test the client driver sends `test-done` message to the server, server closes the session and responds and the driver goes to idle page.
But it's possible that the page is redirected in the moment between `test-done` request is sent but response is not received. In this case on the page after redirect driver sends any request (`error` or `ready`) to the server but gets the error because the session is closed already. 
Take a look to the fix: https://github.com/DevExpress/testcafe/pull/464/commits/8da6fbe04d0ed22e60149c6eeea2113b42fe62f4
Tests are successful three times in a row with this

File: src/client/driver/index.js [hunk 1]
```diff
@@ -39,6 +40,11 @@ export default class ClientDriver {
         modalBackground.initAndShowLoadingText();
         hammerhead.on(hammerhead.EVENTS.uncaughtJsError, err => this._onJsError(err));
 
+        if (this.contextStorage.getItem(TEST_DONE_SENT_FLAG)) {
```
Comment (inikulin) [inline src/client/driver/index.js]: This requires comment which describes why we need this.

Comment (AlexanderMoskovkin) [inline src/client/driver/index.js]:
```js
// NOTE: we should not send any message to the server if we've
// sent the 'test-done' message but haven't got the response.
```
is this clear?

**Comment by inikulin** [comment]
@testcafe-build-bot \retest

**Comment by inikulin** [comment]
@testcafe-build-bot \retest

**Comment by inikulin** [comment]
@testcafe-build-bot \retest

**Comment by inikulin** [comment]
@testcafe-build-bot \retest

**Comment by inikulin** [comment]
@testcafe-build-bot \retest

**Comment by inikulin** [comment]
@testcafe-build-bot \retest

**Comment by inikulin** [comment]
I'm afraid it wasn't fixed

**Comment by AlexanderMoskovkin** [comment]
It's not so bad. This is another type of failing, I've reproduced this manually a few times. I'll fix it

**Comment by inikulin** [comment]
I was talking about this one: https://travis-ci.org/testcafe-build-bot/testcafe/jobs/124314491 (it's latest test run in this PR) and it still fails with the same error

**Comment by AlexanderMoskovkin** [comment]
Yes, I see. The test is the same and the error is the same. But I see in the screencast it's an another problem, believe me =)

**Comment by inikulin** [comment]
lgtm, then =)

**Comment by AlexanderMoskovkin** [comment]
I'll try to fix this problem in this PR too

**Comment by AlexanderMoskovkin** [comment]
@VasilyStrelyaev please check the [comment](https://github.com/DevExpress/testcafe/pull/464/files#diff-7b1afb4c4432a8c9497de3cada3cf78dR43)

**Comment by VasilyStrelyaev** [comment]
lgtm

**Comment by AlexanderMoskovkin** [comment]
So, this PR made functional tests more stable. But there is a problem with them that is reproduced seldom. I create a separate issue for it: #469

**Pull request merged**